app/view_objects/notifications_view.rb

```diff
@@ -0,0 +1,40 @@
+class NotificationsView
+  ORGANISATION_URL_PATTERN = /\/organisations\/(\S+)\/?/.freeze
+
+  attr_reader :user_notification_preferences
+
+  def initialize(
+    request:,
+    current_user:,
+    user_notification_preferences:
+  )
+    @request = request
+    @current_user = current_user
+    @user_notification_preferences = user_notification_preferences
+  end
+
+  def user_id
+    current_user["user_id"]
+  end
+
+  def user_email
+    current_user["info"]["email"]
+  end
+
+  def cancel_link_path
+    return Rails.application.routes.url_helpers.root_path if ORGANISATION_URL_PATTERN.match(request.referer).nil?
+
+    URI(request.referer).path
```

Comment:
does it explode if for some reason if can't determine the referrer? or is that not possible? also i think it's spelt `referrer`?

Reply:
If you go directly to http://localhost:3000/notifications `request.referer` is nil and cancel_link_path is root_path